src/pages/sales-order/Tabs/SalesOrderForm.js

```diff
@@ -0,0 +1,1364 @@
+import CustomDatePicker from 'src/components/Inputs/CustomDatePicker'
+import { formatDateFromApi, formatDateToApi } from 'src/lib/date-helper'
+import { Grid, FormControlLabel, Checkbox } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { SystemFunction } from 'src/resources/SystemFunction'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import { useForm } from 'src/hooks/form'
+import WorkFlow from 'src/components/Shared/WorkFlow'
+import { useWindow } from 'src/windows'
+import { ControlContext } from 'src/providers/ControlContext'
+import {
+  getIPR,
+  DIRTYFIELD_QTY,
+  DIRTYFIELD_BASE_PRICE,
+  DIRTYFIELD_UNIT_PRICE,
+  DIRTYFIELD_MDAMOUNT,
+  DIRTYFIELD_UPO,
+  DIRTYFIELD_EXTENDED_PRICE
+} from 'src/utils/ItemPriceCalculator'
+import { getVatCalc } from 'src/utils/VatCalculator'
+import { getDiscValues, getFooterTotals, getSubtotal } from 'src/utils/FooterCalculator'
+import AddressFilterForm from './AddressFilterForm'
+import { AddressFormShell } from 'src/components/Shared/AddressFormShell'
+
+export default function SalesOrderForm({
+  labels,
+  access: maxAccess,
+  siteId,
+  recordId,
+  defaultSalesTD,
+  currency,
+  plant,
+  salesPerson,
+  dtId,
+  window
+}) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { stack } = useWindow()
+  const { platformLabels } = useContext(ControlContext)
+  const [cycleButtonState, setCycleButtonState] = useState({ text: '%', value: 2 })
+  const [address, setAddress] = useState({})
+
+  const [initialValues, setInitialData] = useState({
```

Comment:
since we are not using setInitialData can't we call them in the formik directly and remove the state?